test-data/unit/check-unsupported.test

```diff
@@ -11,5 +13,5 @@ def g(): pass
 @d # E
 def g(x): pass
 [out]
-main:5: error: Name 'f' already defined
-main:7: error: Name 'g' already defined
+tmp/foo.pyi:7: error: Name 'g' already defined
+tmp/foo.pyi:5: error: Name 'f' already defined
```

Comment:
I spent a while tracking down the reason for it:
* For objects we initially thought were part of an overload series, but turn out not to be, we give the already-defined error in semantic analysis phase two, now.
* For objects that are plain redefinitions, we give the already-defined error in phase 1.
* We were setting an `import_context` object differently between these two phases
* Error sorting groups by `import_context`, not just file.

I've made a change to set `import_context` the same way between the two phases.